spec/relativetimeformat.html

```diff
@@ -1,61 +1,256 @@
-        1. Set _relativeTimeFormat_.[[numberFormat]] to _nf_.format.
-        1. Set _relativeTimeFormat_.[[boundResolve]] to *undefined*.
-        1. Set _relativeTimeFormat_.[[initializedRelativeTimeFormat]] to *true*.
+        1. Set _relativeTimeFormat_.[[Fields]] to _fields_.
+        1. Let _nf_ be Construct(%NumberFormat%, «[_r_.[[Locale]], {minimumIntegerDigits: 2, useGrouping: *false*}»).
+        1. Set _relativeTimeFormat_.[[NumberFormat]] to _nf_.format.
+        1. Let _pr_ be Construct(%PluralRules%, «[_r_.[[Locale]], {style: *"cardinal"*}»).
+        1. Set _relativeTimeFormat_.[[PluralRules]] to _pr_.select.
+        1. Set _relativeTimeFormat_.[[InitializedRelativeTimeFormat]] to *true*.
         1. Return _relativeTimeFormat_.
       </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-PartitionRelativeTimePattern" aoid="PartitionRelativeTimePattern">
+      <h1>PartitionRelativeTimePattern (relativeTimeFormat, x)</h1>
+
+      <p>
+        When the FormatRelativeTime abstract operation is called with arguments _relativeTimeFormat_ (which must be an object initialized as a RelativeTimeFormat) and _value_ (which must be a Number value), it returns a String value representing _value_ (interpreted as a time value as specified in ES2015, 20.3.1.1) according to the effective locale and the formatting options of _relativeTimeFormat_.
+      </p>
+
+      <emu-alg aoid="PartitionRelativeTimePattern">
+        1. If _x_ is not a finite Number, then throw a *RangeError* exception.
+        1. Let _fields_ be _relativeTimeFormat_.[[Fields]].
+        1. Let _style_ be _relativeTimeFormat_.[[Style]].
+        1. Let _now_ be %Date_now%().
+        1. Let _ms_ be _x_ - _now_.
+        1. Let _units_ be ComputeTimeUnits(_ms_).
+        1. If _relativeTimeFormat_.[[Unit]] is equal to "best fit", then
+          1. Let _unit_ be GetBestMatchUnit(_units_).
+        1. Else
+          1. Let _unit_ be _relativeTimeFormat_.[[unit]].
+        1. If _style_ is equal to "short", then
+          1. Let _entry_ be _unit_ + "-short".
+        1. Else if _style_ is equal to "narrow", then
+          1. Let _entry_ be _unit_ + "-narrow".
+        1. Else
+          1. Let _entry_ be _unit_.
+        1. Let _exists_ be ? HasProperty(_fields_, _entry_).
+        1. If _exists_ is *false*, then
+          1. Let _entry_ be _unit_.
+        1. Let _patterns_ be ? Get(_fields_, _entry_).
+        1. Let _v_ be _units_.[[&lt;_unit_&gt;]].
+        1. If _v_ is between -2 (inclusive) and 2 (inclusive), then
+          1. Let _exists_ be ? HasProperty(_patterns_, ToString(_v_)).
+          1. If _exists_ is *true*, then
+            1. Let _result_ be Get(_patterns_, ToString(_v_)).
+            1. Return _result_.
+        1. If _v_ is less than 0, then
+          1. Let _tl_ be "past".
+        1. Else
+          1. Let _tl_ be "future".
+        1. Let _po_ be ? Get(_patterns_, _tl_).
+        1. Let _fv_ be _relativeTimeFormat_.[[NumberFormat]](_v_).
```

Comment:
this call might throw. also, I don't think we can call them directly since they are callable objects.